client.go

```diff
@@ -208,8 +219,40 @@ type batchRequest struct {
 	Batch []Event `json:"batch"`
 }
 
+/* TriggerBatch triggers multiple events on multiple types of channels in a single call:
+    client.TriggerBatch([]pusher.Event{
+	    { Channel: "donut-1", Name: "ev1", Data: "pippo1" },
+	    { Channel: "private-encrypted-secretdonut", Name: "ev2", Data: "pippo2" },
+    })
+*/
 func (c *Client) TriggerBatch(batch []Event) (*BufferedEvents, error) {
-	payload, err := json.Marshal(&batchRequest{batch})
+
+	hasEncryptedChannel := false
+	// validate every channel name and every sockedID (if present) in batch
+	for _, event := range batch {
+
+		if len(event.Channel) > maxChannelNameSize || !channelValidationRegex.MatchString(event.Channel) {
+			return nil, fmt.Errorf("The channel named %s has a non-valid name", event.Channel)
```

Comment:
I think this should re-use the same logic that `channelsAreValid` is using. i.e. define a new function

```go
func validChannel(channel string) bool {
	if len(channel) > maxChannelNameSize || !channelValidationRegex.MatchString(channel) {
		return false
	}
	return true
}
```

Then this can be called both here, and also in by `channelsAreValid`.

Reply:
Oh yes, good idea! 👌